dbms/src/Storages/GCManager.cpp

```diff
@@ -97,8 +97,10 @@ bool GCManager::work()
     }
     if (iter == storages.end())
         iter = storages.begin();
-    next_table_id = iter->first;
-    LOG_DEBUG(log, "End GC and next gc will start with table id: {}", next_table_id);
+
+    if (iter != storages.end())
```

Comment:
This if statement always be true, this line check seems useless.

Reply:
This statement will be false if there is no storage to GC, for example, the TiFlash instance just starts up and there is no schema syncing happening before it receives some r/w request. Otherwise, an undefined value will be logged.